### PR TITLE
Fixed: `TDocVariantData.B` or `GetBooleanByName` couldn't correctly handle the "true" and "false" JSON values.

### PR DESCRIPTION
**Pull request by edwinyzh**

### CODE REVIEW

File: SynCommons.pas [hunk 1]
```diff
@@ -21321,6 +21321,8 @@ function VariantToBoolean(const V: Variant; var Value: Boolean): boolean;
     Value := TVarData(V).VBoolean;
   varInteger: // coming e.g. from GetJsonField()
     Value := TVarData(V).VInteger=1;
+  varString, varUString: // handles "true"/"false" values in JSON
```
Comment (synopse) [inline SynCommons.pas]: varUString should be used only with a proper {$ifdef HASVARUSTRING} {$endif} confitional.

File: SynCommons.pas [hunk 2]
```diff
@@ -21321,6 +21321,8 @@ function VariantToBoolean(const V: Variant; var Value: Boolean): boolean;
     Value := TVarData(V).VBoolean;
   varInteger: // coming e.g. from GetJsonField()
     Value := TVarData(V).VInteger=1;
+  varString, varUString: // handles "true"/"false" values in JSON
+    Value := SameTextU(VariantToUTF8(V), BOOL_UTF8[True]);
```
Comment (synopse) [inline SynCommons.pas]: this will create a temporary string on the stack, which slows down this function a lot (hidden try...finally block)

**Comment by synopse** [comment]
I guess https://github.com/synopse/mORMot/commit/99e8896c0bed259fe37d32436799e8c9e6db4f39 properly implement what you expected

**Comment by edwinyzh** [comment]
> I guess [99e8896](https://github.com/synopse/mORMot/commit/99e8896c0bed259fe37d32436799e8c9e6db4f39) properly implement what you expected

Thanks ab! I was amending the code as per your request above, and it's good you do it by yourself :)
Tested and it works.

**Pull request closed without merging**